src/targets/maven.ts

```diff
@@ -355,14 +391,33 @@ export class MavenTarget extends BaseTarget {
    * @param distDir directory of the distribution.
    * @returns record of required files.
    */
-  private getFilesForMavenPomDist(distDir: string): Record<string, string> {
+  private getFilesForMavenPomDist(distDir: string): Record<string, string | string[]> {
     const moduleName = parse(distDir).base;
-    return {
+    const files = {
       targetFile: join(distDir, this.getTargetFilename(distDir)),
       javadocFile: join(distDir, `${moduleName}-javadoc.jar`),
       sourcesFile: join(distDir, `${moduleName}-sources.jar`),
       pomFile: join(distDir, 'pom-default.xml'),
     };
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}-all.jar`))) {
+      Object.assign(files, { allFile: join(distDir, `${moduleName}-all.jar`)})
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}.module`))) {
+      Object.assign(files, { moduleFile: join(distDir, `${moduleName}.module`) })
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}-metadata.jar`))) {
+      Object.assign(files, { metadataFile: join(distDir, `${moduleName}-metadata.jar`) })
```

Comment:
Instead of using `Object.assign`, can we use the following more-readable syntax?

```js
files['allFile'] = join(...);
```